### DISCUSSION TITLE
Are Operation Monitors for Subscriptions supported?

### DISCUSSION THREAD

**Discussion by tsengia** [ankraft/ACME-oneM2M-CSE#99] [Q&A]
I am trying to create a subscription with an operation monitor as the notification criteria, but ACME is giving me an error, below is my input:
```
{
    "m2m:sub": {
        "enc": {
            "om": {
                "ops": [ 1 ]
            },
            "atr": [
                {
                    "nm": "cnd",
                    "val": "edu.psu.cse.traffic.trafficLightIntersection"
                }
            ]
        },
        "nu": [
            "Cdashboard"
        ],
        "et": "20231107T231239,059849",
        "nct": 1
    }
}
```

The response is a `4000` response code:
```
{
    "m2m:dbg": "Attribute/value validation error: enc={'om': {'ops': [1]}, 'atr': [{'nm': 'cnd', 'val': 'edu.psu.cse.traffic.trafficLightIntersection'}]} (unknown or undefined attribute:om in complex type: m2m:eventNotificationCriteria)"
}
```

**Comment by ankraft** [comment]
The <subscription> resource type is very complex and I haven't implemented all the functionalities yet. Unfortunately, the *operationMonitor* of *eventNotificationCriteria* is not yet supported :(

Is it an important feature for you?

But the error message points to another problem. You are using the *development* branch right now?  There is a misconfiguration in the attribute validation that I just corrected.

Also, from looking at your *atr* structure: this is a list of attribute names (strings) not a complex type.

**Reply by tsengia** [reply]
Don't worry about this for now. I've decided to just use the notification event type for now and it works well.

**Comment by reinaortega** [comment]
Just a comment on om element. It seems to me that ops should not be an array. It's just a number that gives the set of operations desired (m2m:accessControlOperations).
As Andreas said, probably the functionality is not implemented, but the enc can be correct.

**Reply by ankraft** [reply]
TS-0004, 6.3.4.2.29m2m:accessControlOperations, says "Used in \<accessControlPolicy>, filterCriteria and operationMonitor.". Perhaps we should improve this to 

 > Used in \<accessControlPolicy>, m2m:filterCriteria and m2m:operationMonitor.

And yes, starting with the current development branch (0.11.0-beta) complex type attributes are better validated now.
